### DISCUSSION TITLE
Is there is way to listen/ select to only a certain value of a field

### DISCUSSION THREAD

**Discussion by shashkiranr** [rrousselGit/riverpod#2926] [Q&A] [answered]
```
enum Dummy{
one,
two,
three,
}

final dummyProvider = StateProvider<Dummy>((ref) {
  return Dummy.one;
});

// some widget build
build(){

// this should be triggered only when the value of dummyProvider is two
  ref.watch(dummyProvider)
}
```

is this possible using a StateNotifier ? I know we can use select and listen to a field but is there a way to listen to particular value of the field?

**Comment by rrousselGit** [comment, marked as answer]
No. That's generally considered anti pattern.

Because ref.watch wouldn't be able to return anything if the value isn't what you expect it to be

**Reply by shashkiranr** [reply]
I understand that. But is there a way I can specify the value when selecting or listening to the provider and when changed to that particular value the provider notifies.

I have multiple constant widgets which inherit specific build method. All the widgets are listening to a field in StateNotifier or single StateProvider and they rebuild every time a new value is set.  

Something like 

T being equal to the value specified while calling ref.watch(ref.select((value) => value.field, T)

if(state == T) { 
notifyListener();
}

**Reply by rrousselGit** [reply]
Can't you just select "ref.watch(provider.select((v) => v.field == value))"?

If the bool is true, you know it the field is "value"
